src/addrman.cpp

```diff
@@ -756,9 +760,10 @@ std::pair<CAddress, NodeSeconds> AddrManImpl::Select_(bool new_only, std::option
             position = (initial_position + i) % ADDRMAN_BUCKET_SIZE;
             node_id = GetEntry(search_tried, bucket, position);
             if (node_id != -1) {
-                if (network.has_value()) {
+                if (networks.has_value()) {
                     const auto it{mapInfo.find(node_id)};
-                    if (Assume(it != mapInfo.end()) && it->second.GetNetwork() == *network) break;
+                    auto& nets = *networks;
+                    if (Assume(it != mapInfo.end()) && std::find(nets.begin(), nets.end(), it->second.GetNetwork()) != nets.end()) break;
```

Comment:
This code is correct, but is doing linear search in the list of networks. For values of up to 6 that is irrelevant, but if it is changed from `vector` to `unordered_set` then lookup is also shorter to write, in addition to being `O(1)`:

```suggestion
                    if (Assume(it != mapInfo.end()) && networks->contains(it->second.GetNetwork())) break;
```

Reply:
Nice, I'll address it.